### PR TITLE
cast strings

### PR DESCRIPTION
**Pull request by ahmadabdalla**
## Pipeline Reference

| Pipeline |
| -------- |
|          |

## Type of Change

- [ ] Update to CI Environment or utlities (Non-module effecting changes)
- [ ] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] I have run `Set-AVMModule` locally to generate the supporting module files.
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: avm/res/network/network-manager/connectivity-configuration/main.bicep [hunk 1]
```diff
@@ -49,12 +41,20 @@ resource connectivityConfiguration 'Microsoft.Network/networkManagers/connectivi
...
+        useHubGateway: string(group.useHubGateway) ?? 'false'
+      }
+    )
     connectivityTopology: connectivityTopology
-    deleteExistingPeering: connectivityTopology == 'HubAndSpoke' ? deleteExistingPeering : 'False'
+    deleteExistingPeering: connectivityTopology == 'HubAndSpoke' ? string(deleteExistingPeering) : 'false'
     description: description ?? ''
     hubs: connectivityTopology == 'HubAndSpoke' ? hubs : []
-    isGlobal: isGlobal
+    isGlobal: string(isGlobal) ?? 'false'
```
Comment (AlexanderSehr) [inline avm/res/network/network-manager/connectivity-configuration/main.bicep:57]:
```suggestion
    isGlobal: string(isGlobal)
```

Would that not be enough with the default being `false` anyways?

Comment (ahmadabdalla) [reply avm/res/network/network-manager/connectivity-configuration/main.bicep:57]: Yes it should. Thanks

**Comment by AlexanderSehr** [comment]
Looks good to me @ahmadabdalla, what are your thoughts?

**Comment by ahmadabdalla** [comment]
@AlexanderSehr yeah I am happy with the current implementation but I am getting some deployment failures that I need to diagnose. Essentially the failures are all due to references to nonexistent resources although they are specified properly like "networkManager.id".. I am thinking to implement an explicit depends on. Something tells me we have seen these problems in the past

**Comment by AlexanderSehr** [comment]
> @AlexanderSehr yeah I am happy with the current implementation but I am getting some deployment failures that I need to diagnose. Essentially the failures are all due to references to nonexistent resources although they are specified properly like "networkManager.id".. I am thinking to implement an explicit depends on. Something tells me we have seen these problems in the past

That rather sounds like an ICM issue I opened a while back. Feel free to add your observations here: https://github.com/Azure/bicep/issues/13457

**Pull request merged**